### ISSUE TITLE
SlevomatCodingStandard.Namespaces.UnusedUses with searchAnnotations false positive where Annotation defines intersection type

### ISSUE THREAD

**Issue by BinaryKitten** [slevomat/coding-standard#1262]
When using the `SlevomatCodingStandard.Namespaces.UnusedUses` in our standards file, we get an error even though we have the item defined as use.

We're currently limited to PHP 7.3 - so using annotations to backup our basic property usages - this is mostly used in our unit tests.

Standards item:
```xml
<rule ref="SlevomatCodingStandard.Namespaces.UnusedUses">
    <properties>
        <property name="searchAnnotations" type="bool" value="true"/>
    </properties>
</rule>
```
Failing item
```php
<?php

declare(strict_types=1);

namespace Tests\Unit;

use App\Http\Controllers\Controller;
use PHPUnit\Framework\MockObject\MockObject;
use PHPUnit\Framework\TestCase;

class FailingIntersectionTest extends TestCase
{
    /** @var null|Controller&MockObject **/
    protected $mockController = null;

    protected function setUp(): void
    {
        $this->mockController = $this->createMock(Controller::class);
    }

    public function testExampleNotReallyATest(): void
    {
        $this->assertTrue(true);
    }
}
```

Outcome
```
E 1 / 1 (100%)

FILE: /srv/www/application/laravel/tests/Unit/FailingIntersectionTest.php
-------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------------------------------------------------------
 8 | ERROR | [x] Type PHPUnit\Framework\MockObject\MockObject is not used in this file.
   |       |     (SlevomatCodingStandard.Namespaces.UnusedUses.UnusedUse)
-------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-------------------------------------------------------------------------------------------------------------------------------------------------
```

**Comment by kukulich** [comment]
@BinaryKitten what version of slevomat cs do you use?

**Comment by kukulich** [comment]
@BinaryKitten btw there may be problem that you have invalid end of the doccomment, there should be just on asterisk.

**Comment by BinaryKitten** [comment]
@kukulich version 7.0.13

as to the extra asterisk at the end, 

same happens with:
```php
/** @var null|Controller&MockObject */
```
and
```php
    /**
     * @var null|Controller&MockObject
     */
```
and
```
/**
 * @var null|Controller&MockObject
 **/
 ```

**Comment by kukulich** [comment]
You have to use this version: `@var null|(Controller&MockObject)`
